app/scripts/services/serviceInstances.js

```diff
@@ -3,64 +3,105 @@
 
-        return DataService.delete({
-            group: 'servicecatalog.k8s.io',
-            resource: 'serviceinstances'
-          },
-          apiObject.metadata.name,
-          { namespace: apiObject.metadata.namespace },
-          { propagationPolicy: null
-          }) // TODO - remove once this is resolved https://github.com/kubernetes-incubator/service-catalog/issues/942
-          .then(function() {
-            NotificationsService.addNotification({
-              type: "success",
-              message: "Successfully deleted provisioned service " + apiObject.metadata.name + "."
-            });
-          }, function(err) {
-            NotificationsService.addNotification({
-              id: "deprovision-service-error",
-              type: "error",
-              message: "An error occurred while deleting provisioned service " + apiObject.metadata.name + ".",
-              details: getErrorDetails(err)
-            });
+        var context = {namespace: apiObject.metadata.namespace};
+        var resource = {
+          group: 'servicecatalog.k8s.io',
+          resource: 'serviceinstances'
+        };
+
+        // TODO - remove once this is resolved https://github.com/kubernetes-incubator/service-catalog/issues/942
+        var opts = {
+          propagationPolicy: null
+        };
+
+        return DataService.delete(resource, apiObject.metadata.name, context, opts).then(function() {
+          NotificationsService.addNotification({
+            type: "success",
+            message: "Successfully deleted provisioned service " + apiObject.metadata.name + "."
+          });
+        }, function(err) {
+          NotificationsService.addNotification({
+            id: "deprovision-service-error",
+            type: "error",
+            message: "An error occurred while deleting provisioned service " + apiObject.metadata.name + ".",
+            details: getErrorDetails(err)
           });
+        });
       });
-    }
+    };
+
+    var showUnableToDeprovisionDialog = function() {
+      return $uibModal.open({
+        animation: true,
+        templateUrl: 'views/modals/warning.html',
+        controller: 'WarningModalController',
+        resolve: {
+          modalConfig: function() {
+            return {
+              title: "Unable to Deprovision",
+              strongMessage: "This service cannot be deprovisioned because it has bindings.",
+              message: "You must delete all bindings from the service before you can deprovision it."
+            };
+          }
+        }
+      });
+    };
+
+    var deprovision = function (apiObject) {
+      var context = {namespace: apiObject.metadata.namespace};
+      var resource = {
+        group: 'servicecatalog.k8s.io',
+        resource: 'serviceinstancecredentials'
+      };
+
+      if (CatalogService.SERVICE_CATALOG_ENABLED) {
+        DataService.list(resource, context, function (bindings) {
```

Comment:
I believe everywhere we call this, we already have a list of the bindings. I'd rather just pass those in if we have them.

Reply:
The service instance page does not have the bindings. I can take them optionally and only retrieve if not given.

Reply:
> The service instance page does not have the bindings.

Hey @jeff-phillips-18 I'm OK with it in that case.

I want to look at the instance page anyway later. It looks like it's requesting service classes twice, and that's potentially a lot of data.

In general, I think it's better for the page controllers to get the data and pass them down to the components to avoid requesting things more than once.

Reply:
I can take a look at that as part of this. Basically its the ResourceServiceBindings that gets the bindings for the service instance page (an all the 'application' pages. I can update that logic as well if you'd like.

Reply:
It might be better to wait for #2155 since I have some changes to instance page and resource service bindings component in flight :/

Reply:
Pushed changes to only get bindings when not passed. Holding off on updating all the controllers using the ResourceServiceBindings.